themes/powerline/powerline.base.bash

```diff
@@ -99,7 +99,7 @@ function __powerline_scm_prompt {
     elif [[ "${SCM_HG_CHAR}" == "${SCM_CHAR}" ]]; then
       scm_prompt+="${SCM_CHAR}${SCM_BRANCH}${SCM_STATE}"
     fi
-    echo "${scm_prompt}${scm}|${color}"
+    echo "$(eval "echo ${scm_prompt}")${scm}|${color}"
```

Comment:
Could this be a security risk? Could this be used to execute malicious code, e.g. by creating a branch name in a special format?

Reply:
Whilst this seems possible, I'm not sure if this would happen? I'm willing to be corrected with a counter example, but from the standpoint of arbitrary command execution I think this _should_ be safe.

To the best of my ability, it should be evaluating `${scm_prompt}` to `${_git-friendly-ref}` in the outer quotes, which is then being evaluated by the quoted echo command in the sub-shell `eval`.

A small (but by no means comprehensive) test:
``` shell
~:$ mkdir /tmp/git-test/; cd /tmp/git-test/
/tmp/git-test/:$ git init
/tmp/git-test/:$ touch foobar
/tmp/git-test/ $ ls
foobar
/tmp/git-test/:$ git checkout -b ls
```
![image](https://user-images.githubusercontent.com/7091722/54356340-4aa3f600-4653-11e9-8ef8-cdcbdd976fd2.png)

I also attempted to write a little script that has the side effect of creating a file to see if I could achieve side effects, and even when the branch name was set to `script.sh` (the name of the script), it didn't seem to run. Slashes and spaces are not valid git branch names, so more complex commands or paths _shouldn't_ work either. 

Please correct me if I am wrong though.

Reply:
Sounds good - thanks for the thorough evaluation!